bhamcal/extractor.py

```diff
@@ -69,3 +69,20 @@ def extract_datetime(date, time):
     dt = DEFAULT_TIMEZONE.localize(dt)
     dt = dt.astimezone(pytz.utc)
     return dt
+
+
+def clean_subject(subject: str) -> str:
+    # Was not needed in my case but I'll assume others do.
+    one = r"^LI "
+    # Remove module codes, LM and/or LH and extended.
+    two = r"(?P<code>\([0-9]+/[0-9]+\))|(?P<prefix>LM/LH|LH/LM|LH|LM)|(?P<extended>\(Extended\)?)"
+    # Remove duplicates in the case of the name being present twice on some extended modules.
+    three = r"(?P<one>^.*/)"
+    round_one_re = re.compile(one,re.MULTILINE)
+    round_two_re = re.compile(two, re.MULTILINE)
```

Comment:
nit: compile these outside of the function

Reply:
Cheers forgot about that.